src/zcash/JoinSplit.cpp

```diff
@@ -235,6 +236,10 @@ class JoinSplitCircuit : public JoinSplit<NumInputs, NumOutputs> {
             out_macs[i] = PRF_pk(inputs[i].key, i, h_sig);
         }
 
+        if (!computeProof) {
+            return boost::array<unsigned char, ZKSNARK_PROOF_SIZE>();
```

Comment:
Does this zero-initialize? [Yes it does](http://stackoverflow.com/a/18308795) because there are `()` there. It's somewhat surprising that in C++ the person instantiating the object decides whether members get automatically initialized with a difference of 2 characters.

(e.g. if we did

```
boost::array<unsigned char ZKSNARK_PROOF_SIZE> ret;
retur ret;
```

then it would be uninitialized memory).